packages/fxa-auth-server/lib/luaScripts/setAccessToken_2.lua

```diff
@@ -1,11 +1,16 @@
 local userId = KEYS[1]
 local tokenId = KEYS[2]
 local token = ARGV[1]
-local ttl = ARGV[2]
+local limit = tonumber(ARGV[2])
+local ttl = ARGV[3]
 local args = {token}
 if ttl then
   table.insert(args, 'PX')
   table.insert(args, ttl)
 end
+local count = redis.call('scard', userId)
+if count > limit then
+  redis.call('spop', userId, math.min((count - limit) + math.floor(limit / 10), limit))
+end
```

Comment:
I wasn't following your comment "It makes no effort to check expiry so in the worst case up to 10%
of the limit active tokens will be deindexed, but still active."

But, as an example, if limit = 100, and count = 180, then Math.min(180 - 100 + 10, 100) = 90 tokens, of 50% are removed. So how does the 10% number fit with these?

Reply:
10% in the steady state case, where count never gets over 101. in the initial overflow case it'll remove up to `limit` so yes, 90 in your case.

Reply:
Ah, ok.